src/Runner.Worker/JobExtension.cs

```diff
@@ -239,7 +239,18 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
                     // Download actions not already in the cache
                     Trace.Info("Downloading actions");
                     var actionManager = HostContext.GetService<IActionManager>();
-                    var prepareResult = await actionManager.PrepareActionsAsync(context, message.Steps);
+                    PrepareResult prepareResult;
+                    // TODO should we also check DistributedTask.NewActionMetadata, 
+                    // we removed the backcompat code, so if NewActionMetadata was disabled, and Composite enabled, we would run into issues
+                    // TODO Make a real feature flag
+                    if (!string.IsNullOrEmpty(context.Global.Variables.Get("ENABLE_COMPOSITE")))
```

Comment:
Rather than permanently leave a variable (`ENABLE_COMPOSITE`, `NEW_ACTION_METADATA`), we may want to leverage `PlanUtil.GetFeatures`. We can probably correlate plan version to GHES version.

Think about it. Sort of on the fence either way.